rag_experiment_accelerator/doc_loader/tests/test_pdf_loader.py

```diff
@@ -0,0 +1,33 @@
+from unittest.mock import Mock
+
+import pytest
+from rag_experiment_accelerator.doc_loader.pdfLoader import load_pdf_files
+
+
+@pytest.mark.parametrize(
+    "pypdf_enabled, expected_content",
+    [
+        (True, "this is a sample pdf with text."),
+        (
+            False,
+            "this is a sample of text captured as an image and stored as pdf.this is a sample pdf with text.",
+        ),
+    ],
+)
+def test_load_pdf_files(pypdf_enabled: bool, expected_content: str):
+    file_path = "./data/pdf/text-as-image.pdf"
+    chunk_size = 1000
+    overlap_size = 400
+
+    environment_mock = Mock(spec=["pypdf_enabled"])
```

Comment:
Severity: Potential Issue - The `Mock` object is being created with a spec that only includes 'pypdf_enabled'. If there are other attributes or methods that `load_pdf_files` might access on the environment object, consider including them in the spec to avoid potential issues during testing.